osf/utils/fields.py

```diff
@@ -12,7 +12,7 @@ class LowercaseCharField(models.CharField):
     def get_prep_value(self, value):
         value = super(models.CharField, self).get_prep_value(value)
         if value is not None:
-            value = value.lower()
+            value = str(value).lower()
```

Comment:
Hmm. Anyone see why, following that path, they wouldn't come out the other side as strings? You remember what type of thing you were testing with @mfraezz?